File: 0289-game-of-life/0289-game-of-life.cpp
class Solution {
public:
    void gameOfLife(vector<vector<int>>& board) {
        for(int i=0;i<board.size();i++){
            for(int j=0;j<board[0].size();j++){
                transform(board, i, j);
            }
        }
        for(int i=0;i<board.size();i++){
            for(int j=0;j<board[0].size();j++){
                if(board[i][j]==2) board[i][j] = 0;
                if(board[i][j]==3) board[i][j] = 1;
            }
        }
    }
    
    int dir[9] = {-1,0,1,0,-1,-1,1,1,-1};
    
    void transform(vector<vector<int>>& board, int i, int j){
        int count =0;
        for(int d=0;d<8;d++){
            int x=i+dir[d],y=j+dir[d+1];
            if(x<0||y<0||x>=board.size()||y>=board[0].size()) continue;
            if(board[x][y]==1||board[x][y]==2) count++;
        }
        if(board[i][j]==0 && count==3) board[i][j] = 3;
        if(board[i][j]==1 && (count<2 || count>3)) board[i][j]=2;
    }
};

File: 0289-game-of-life/0289-game-of-life.py
class Solution:
    def gameOfLife(self, board: List[List[int]]) -> None:
        for i in range(len(board)):
            for j in range(len(board[0])):
                transform(board,i,j)
        for i in range(len(board)):
            for j in range(len(board[0])):
                if board[i][j]==2:
                    board[i][j]=0
                if board[i][j]==3:
                    board[i][j]=1

def transform(board, i, j):
    dir = [-1,0,1,0,-1,-1,1,1,-1]
    count = 0
    for d in range(8):
        x = i+dir[d]
        y = j+dir[d+1]
        if x>=0 and y>=0 and x<len(board) and y<len(board[0]):
            if board[x][y]==1 or board[x][y]==2:
                count+=1
    if board[i][j]==0 and count==3:
        board[i][j] = 3
    if board[i][j]==1 and (count<2 or count>3):
        board[i][j] = 2